### DISCUSSION TITLE
Error building MIlvus on a local OS/shell environment

### DISCUSSION THREAD

**Discussion by ZubeenK** [milvus-io/milvus#24491] [Q&A and General discussion]
I have followed the steps provided in the link: https://github.com/milvus-io/milvus/blob/master/DEVELOPMENT.md#building-milvus-with-docker, to build Milvus on a local OS. I have the following versions installed from prerequisites list: 
gcc = 7.5.0
go = go1.20.4 linux/amd64
cmake = 3.26.3
make = (GNU Make 4.2.1)
conan = 1.58.0

To build Milvus, when I run the command: make, I am getting the following error: 
*** No rule to make target 'install'.  Stop.
make: *** [Makefile:140: build-cpp] Error 1

![image](https://github.com/milvus-io/milvus/assets/73072849/b44b99a6-7e05-4916-af6d-ac9467d7c797)

While running the make command, I am also getting the below error:
![image](https://github.com/milvus-io/milvus/assets/73072849/715bc506-bfc7-4013-bf20-8d42306ad4dc)

Can anyone help me with this issue?

**Comment by yhmo** [comment]
gcc = 7.5.0 version is low. 
My OS is Ubuntu 20.04, the gcc version is v9.4, I can build the project successfully.

**Reply by xiaofan-luan** [reply]
did you run ./scripts/install_deps.sh?

seems it lacks boost and maybe other dependency libraries

**Reply by ZubeenK** [reply]
@xiaofan-luan , yes I ran the command: ./scripts/install_deps.sh, but still I am getting the error.

**Reply by ZubeenK** [reply]
@yhmo the link: https://github.com/milvus-io/milvus/blob/master/DEVELOPMENT.md#building-milvus-with-docker mentioned gcc version as 7.5.0, so I installed the specific version, and then I am getting this error.

**Reply by yhmo** [reply]
> @yhmo the link: https://github.com/milvus-io/milvus/blob/master/DEVELOPMENT.md#building-milvus-with-docker mentioned gcc version as 7.5.0, so I installed the specific version, and then I am getting this error.

I think the doc is out of date. Last month I tried gcc 7.5 on ubuntu 18 but failed to build the project. Then I upgraded my OS to ubuntu 20 and succeed.

**Reply by ZubeenK** [reply]
@yhmo I tried to build using gcc 9.4, ubuntu 20, now I am getting the following error:
![image](https://github.com/milvus-io/milvus/assets/73072849/55970239-bbf5-4ed1-9131-4211d3807cb9)
Any idea how I can resolve this?

**Reply by yhmo** [reply]
> @yhmo I tried to build using gcc 9.4, ubuntu 20, now I am getting the following error: ![image](https://user-images.githubusercontent.com/73072849/242504015-55970239-bbf5-4ed1-9131-4211d3807cb9.png) Any idea how I can resolve this?

Maybe the boost lib version is old.

Use this command to check boost version under Ubuntu:
`dpkg -S /usr/include/boost/version.hpp`

My boost version is: libboost1.71-dev

**Reply by ZubeenK** [reply]
@yhmo Thanks for replying, I checked the version of libboost and it seems I have the same version of boost: libboost1.71-dev:amd64: /usr/include/boost/version.hpp.

**Reply by yhmo** [reply]
**dynamic_bitset.hpp** is included by `milvus/internal/core/thirdparty/boost_ext/dynamic_bitset_ext.hpp`

Does the file **dynamic_bitset.hpp** exist under the path **/usr/include/boost/** ?
If the file exists, that indicates the boost path was not included correctly. Try search "include boost path" on the Google.
